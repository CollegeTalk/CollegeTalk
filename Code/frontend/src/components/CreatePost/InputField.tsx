import { Dispatch, SetStateAction, forwardRef } from "react";
import { StyleSheet, Keyboard } from "react-native";
import { Input } from "@rneui/themed";

import { primaryColors } from "../../constants/Colors";

const styles = StyleSheet.create({
    inputWrapper: {
        paddingHorizontal: 0
    },
    inputContainer: {
        borderBottomWidth: 0
    }
});

type InputFieldProps = {
    showError: boolean;
    errorMessage?: string;
    placeholder: string;
    setText: Dispatch<SetStateAction<string>>;
    toggleError?: Dispatch<SetStateAction<boolean>>;
    isLarge: boolean;
};

const InputField = forwardRef(
    (
        {
            showError,
            errorMessage,
            placeholder,
            setText,
            toggleError,
            isLarge
        }: InputFieldProps,
        ref
    ) => (
        <Input
            ref={ref}
            containerStyle={styles.inputWrapper}
            inputContainerStyle={styles.inputContainer}
            inputStyle={{
                width: "100%",
                height: isLarge ? 100 : 50,
                color: "white",
                borderRadius: 15,
                borderColor: primaryColors.text,
                borderWidth: 1,
                paddingHorizontal: 15,
                /* cannot use paddingVertical for multiline */
                paddingTop: 10,
                paddingBottom: 10,
                marginBottom: 20,
                textAlignVertical: "top"
            }}
            placeholder={placeholder}
            onChangeText={(value: string) => {
                if (toggleError) {
                    toggleError(value === "");
                }
                setText(value);
            }}
            shake={() => showError}
            renderErrorMessage={false}
            errorStyle={{ color: "red" }}
            errorMessage={showError ? errorMessage : undefined}
            multiline={isLarge}
            returnKeyType={!isLarge ? "next" : "default"}
            blurOnSubmit
            onSubmitEditing={() => {
                Keyboard.dismiss();
            }}
        />
    )
);

export default InputField;
