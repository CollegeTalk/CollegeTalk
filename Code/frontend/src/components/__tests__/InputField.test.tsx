import renderer from "react-test-renderer";

import InputField from "../CreatePost/InputField";

describe("<CreatePostTitle />", () => {
    it("has 1 child", () => {
        const tree = renderer
            .create(
                <InputField
                    placeholder=""
                    setText={() => null}
                    isLarge={false}
                />
            )
            .toJSON() as renderer.ReactTestRendererJSON;
        expect(tree.children!.length).toBe(1);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <InputField
                    placeholder=""
                    setText={() => null}
                    isLarge={false}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
