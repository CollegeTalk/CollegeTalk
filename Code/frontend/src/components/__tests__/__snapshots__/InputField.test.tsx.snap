// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePostTitle /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onChangeText={[Function]}
        placeholder=""
        placeholderTextColor="#86939e"
        rejectResponderTermination={true}
        shake={[Function]}
        style={
          Object {
            "borderColor": "#e2d6aa",
            "borderRadius": 15,
            "borderWidth": 1,
            "color": "white",
            "flex": 1,
            "fontSize": 18,
            "height": 50,
            "minHeight": 40,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "textAlignVertical": "top",
            "width": "100%",
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;
